R/calculateJSD.R

```diff
@@ -56,42 +56,42 @@
 #' library(scater)
 #'
 #'
-#' jsd <- calculateJSD(enterotype)
+#' jsd <- getJSD(enterotype)
 #' class(jsd)
 #' head(jsd)
 #'
-#' enterotype <- runMDS(enterotype, FUN = calculateJSD, name = "JSD",
+#' enterotype <- runMDS(enterotype, FUN = getJSD, name = "JSD",
 #'                      exprs_values = "counts")
 #' head(reducedDim(enterotype))
 #' head(attr(reducedDim(enterotype),"eig"))
 #' attr(reducedDim(enterotype),"GOF")
 NULL
 
-setGeneric("calculateJSD", signature = c("x"),
+setGeneric("getJSD", signature = c("x"),
            function(x, ...)
-             standardGeneric("calculateJSD"))
+             standardGeneric("getJSD"))
 
-#' @rdname calculateJSD
+#' @rdname getJSD
 #' @export
-setMethod("calculateJSD", signature = c(x = "ANY"),
+setMethod("getJSD", signature = c(x = "ANY"),
     function(x, ...){
-        .calculate_distance(x, FUN = runJSD, ...)
+        .getJSD(x, ...)
     }
 )
 
-#' @rdname calculateJSD
+#' @rdname getJSD
 #'
 #' @importFrom SummarizedExperiment assay
 #'
 #' @export
-setMethod("calculateJSD", signature = c(x = "SummarizedExperiment"),
+setMethod("getJSD", signature = c(x = "SummarizedExperiment"),
     function(x, assay.type = assay_name, assay_name = exprs_values, 
              exprs_values = "counts", transposed = FALSE, ...){
         mat <- assay(x, assay.type)
         if(!transposed){
             mat <- t(mat)
         }
-        calculateJSD(mat, ...)
+        getJSD(mat, ...)
     }
 )
 
```

Comment:
This generic functions cause quite a lot overhead. Can we just remove them since they are called from getDissimilarity (the data should be ready for .getJSD)?